src/lib.rs

```diff
@@ -490,6 +523,41 @@ mod test {
         assert_eq!(peers_connected(&other_bitcoind.client), 1);
     }
 
+    #[test]
+    fn test_data_persistence() {
+        // Create a Conf with staticdir type
+        let mut conf = Conf::default();
+        conf.staticdir = Some("./bitcoind".into());
```

Comment:
this directory should be a tempdir even if used as staticdir  (by the `let dir` binding it will be deleted at the end of the test, not at the end of the first session)

```suggestion
        let dir = TempDir::new()?;
        conf.staticdir = Some(dir);
```

and remove `         std::fs::remove_dir_all("./bitcoind").unwrap();`

Reply:
Done..